tests/utilities.py

```diff
@@ -0,0 +1,104 @@
+'''
+Module utilities, which provides objects for testing
+'''
+
+import pickle
+
+
+def anthology(list_of_paths_to_files_with_english_texts) -> str:
+    '''
+    Provides an anthology of English texts
+
+    Keyword arguments:
+        list_of_paths_to_files_with_english_texts
+
+    Return values:
+        anthology_of_english_texts
+
+    Side effects:
+        none
+
+    Exceptions raised:
+        none
+
+    Restrictions on when this method can be called:
+        This function will be called by one or more tests.
+    '''
+
```

Comment:
I see you're more intentional in your use of exceptions and asserts, and put more thought into where you use them.  That's totally ok.  I still feel an assert here or there, like in this case `assert isinstance(list_of_paths_to_files_with_english_texts, List), f"Expected list argument but got {type(list_of_paths_to_files_with_english_texts}"` would still be helpful here.

Often in comments you state the type, for example
```
Keyword arguments:
    paths_to_files,   type: List.  Path to files with English text
```
Although this is not gospel.

Ah, yes, I think it's called "Hungarian Notation", where you include the type of an object in the variable name.  Like you did here with `list_`.  I personally do this sometimes in my own personal projects.  However I find pushback from other devs at work.

So ya, there's more than one way to keep things labeled.  And of course python3 allows for calling out the types, although it doesn't enforce it.  Anyway, just wanted to comment on it.  Especially since I like the hungarian notation, but do expect some devs that will be repelled by it 😬

Reply:
I also see your use of asserts in other functions, so there was probably a reason why you didn't include one like I mentioned above.  So ya, once you get the hang of it you can be more selective.  What I mentioned in class is really about building a habit.  If you're starting out it's best to build a habit of using asserts and back off when you are comfortable, as opposed to just always chasing your tail.  I think you're programming at a more advanced level than the others so just wanted to put the comment in context.

Reply:
@EfrainOlivaresUVA , I added the assertion that you suggested. I state the type of keyword arguments when I know or can easily figure out the type. Sometimes I'm not interested in spending time figuring out the type.

I feel refreshed that you use Hungarian notation sometimes. Python doesn't seem to have strong typing or the ability to hover over objects and see their types. I think using Hungarian notation allows me to keep track of what objects really are. If someone doesn't use Hungarian notation I often feel uncertain about their code. I have felt tension with bosses and coworkers who do not use Hungarian notation.

Is there a difference between highways and a system of highways? My mentor Chip Staples asserts no.

Reply:
@EfrainOlivaresUVA , I guess I haven't used asserts generally on keyword arguments but more on results of retrieving or processing information.